packages/vue/src/composables/useAuth.ts

```diff
@@ -1,43 +1,55 @@
+import { createSharedComposable } from '@vueuse/core';
+import { reactive, watchEffect, onScopeDispose } from 'vue';
 import { useActor } from '@xstate/vue';
-import { ref, reactive, Ref, watchEffect } from 'vue';
-import { getServiceFacade } from '@aws-amplify/ui';
+import { interpret } from 'xstate';
+
+import {
+  AuthInterpreter,
+  createAuthenticatorMachine,
+  defaultAuthHubHandler,
+  getServiceFacade,
+  listenToAuthHub,
+} from '@aws-amplify/ui';
+
+import { UseAuth } from '../types';
 import { facade } from './useUtils';
-import { InterpretService } from '@/components';
-
-const service = ref() as Ref<InterpretService>;
-const useAuthenticatorValue = reactive({
-  ...facade,
-  send: '' as unknown,
-  state: '' as unknown,
-}) as any;
-
-export const useAuth = (serv?: InterpretService) => {
-  if (serv) {
-    service.value = serv;
-  }
-  return useActor(service.value);
-};
-
-const useInternalAuthenticator = () => {
-  createValues();
-  watchEffect(() => {
-    createValues();
-  });
-  return useAuthenticatorValue;
-};
 
-export const useAuthenticator = useInternalAuthenticator;
+export const useAuth = createSharedComposable((): UseAuth => {
+  const machine = createAuthenticatorMachine();
+  const service: AuthInterpreter = interpret(machine).start();
+
+  const { state, send } = useActor(service);
+
+  const unsubscribeHub = listenToAuthHub(service, defaultAuthHubHandler);
 
-function createValues() {
-  if (!service.value) return;
+  onScopeDispose(() => {
+    unsubscribeHub();
+  });
+
+  return { service: service, send, state };
+});
 
+export const useAuthenticator = createSharedComposable(() => {
```

Comment:
Using `createSharedComposable` is *not* a breaking change, because we've been relying on the top level global `useAuthenticator` ref value to share the return value.